installation-manager-server-war/src/main/java/com/codenvy/im/service/ReportSender.java

```diff
@@ -46,38 +55,59 @@
  */
 @Singleton
 public class ReportSender {
-    private final LdapManager      ldapManager;
-    private final MailSenderClient mailClient;
-    private final HttpTransport    httpTransport;
-    private final String           updateServerEndpoint;
-    private final ConfigManager    configManager;
+    private static final Logger LOG = Logger.getLogger(ReportSender.class.getSimpleName());
+
+    private final LdapManager               ldapManager;
+    private final MailSenderClient          mailClient;
+    private final HttpTransport             httpTransport;
+    private final String                    updateServerEndpoint;
+    private final ConfigManager             configManager;
+    private final InstallationManagerFacade facade;
 
     @Inject
     public ReportSender(@Named("installation-manager.update_server_endpoint") String updateServerEndpoint,
                         LdapManager ldapManager,
                         MailSenderClient mailClient,
                         HttpTransport httpTransport,
-                        ConfigManager configManager) {
+                        ConfigManager configManager,
+                        InstallationManagerFacade facade) {
         this.ldapManager = ldapManager;
         this.mailClient = mailClient;
         this.httpTransport = httpTransport;
         this.updateServerEndpoint = updateServerEndpoint;
         this.configManager = configManager;
+        this.facade = facade;
     }
 
-//    @ScheduleCron(cron = "0 0 1 ? * SUN *")  // TODO [ndp] CDEC-376
-    // @ScheduleCron(cron = "0 0/1 * 1/1 * ? *")  // send every 1 minute
-    public void sendWeeklyReports() throws IOException, MessagingException, JsonParseException {
-        sendNumberOfUsers();
+    @ScheduleCron(cron = "0 0 1 ? * SUN *")  // send each Sunday at 1:00 AM
+//    @ScheduleCron(cron = "0 0/1 * 1/1 * ? *")  // send every 1 minute
+    public void sendWeeklyReports() {
+        try {
+            Artifact cdecArtifact = getCdecArtifact();
+            Version version = cdecArtifact.getInstalledVersion().orElseThrow(() -> new UnknownArtifactVersionException(cdecArtifact));
+
+            if (!version.is4Major()) {
+                return;
+            }
+
+            sendNumberOfUsers();
+        } catch (JsonParseException | IOException | MessagingException | UnknownArtifactVersionException e) {
+            LOG.log(Level.SEVERE, "Error of sending weekly reports.", e);
+        }
     }
 
     private void sendNumberOfUsers() throws IOException, MessagingException, JsonParseException {
-        ReportParameters parameters = obtainReportParameters(ReportType.CODENVY_ONPREM_USER_NUMBER_REPORT);
-
-        if (! parameters.isActive()) {
-            return;
+        try {
+            if (!facade.isLicenseExpired()) {
+                return;
+            }
+        } catch (LicenseException e) {
+            // log info and send weekly report if there is a problem with Codenvy license
```

Comment:
It is not clear that you are sending weekly report here. Also you send email even when exception doesn't occur. Maybe it would be better to move comment about sending email lower or just remove it.